microcosm_pubsub/dispatcher.py

```diff
@@ -30,19 +30,38 @@ def __init__(self, graph):
         self.sqs_message_context = graph.sqs_message_context
         self.sqs_message_handler_registry = graph.sqs_message_handler_registry
         self.send_metrics = graph.pubsub_send_metrics
+        self.send_batch_metrics = graph.pubsub_send_batch_metrics
         self.max_processing_attempts = graph.config.sqs_message_dispatcher.message_max_processing_attempts
 
     def handle_batch(self, bound_handlers) -> List[MessageHandlingResult]:
         """
         Send a batch of messages to a function.
 
         """
-        instances = [
-            self.handle_message(message, bound_handlers)
-            for message in self.sqs_consumer.consume()
-        ]
+        with elapsed_time(self.opaque):
+            instances = [
+                self.handle_message(message, bound_handlers)
+                for message in self.sqs_consumer.consume()
+            ]
+
+        message_count = len([
+            instance for instance in instances
+            if instance.result != MessageHandlingResultType.IGNORED
+        ])
+
+        self.logger.info(
```

Comment:
Logs and tracks message batch handling outside of a single request-id. This may be less than useful, since the logs will intermingle with different runs of a single handler. It should still be possible to narrow down by tag.

Reply:
Yeah, at the very least I don't think we should put the `elapsed_time` here in `self.opaque`.  For the short term I'd just use a local variable to this handler, but we can think about what we want to do more system wide as we aren't finding the similar problem elsewhere.

Reply:
Fair enough.